tests/utils/test_client.py

```diff
@@ -53,4 +58,41 @@ def test_client_enforce_healthy():
     assert my_client.port == int(CONFIG["my_instance_name"]["port"])
     assert my_client.protocol == CONFIG["my_instance_name"]["protocol"]
     assert my_client.base_path == "/api/versioned/v1/"
-    assert my_client.auth == UsernamePasswordAuth("admin", os.environ["TAMR_TOOLBOX_PASSWORD"],)
+    assert my_client.auth == UsernamePasswordAuth(
+        "admin", os.environ["TAMR_TOOLBOX_PASSWORD"],
+    )
+
+
+@mock_api()
+def test_get_with_connection_retry():
+    # To ensure timeout error is raised to test correct ConnectionError is caught
+
+    log_prefix = "caught_connection_error"
+    # creates temporary file directory & fetches file inside dir
+    with tempfile.TemporaryDirectory() as tempdir:
+        log_file_path = os.path.join(
+            tempdir, f"{log_prefix}_{utils.logger._get_log_filename()}"
+        )
+        log_file_path_extended = os.path.join(
+            log_file_path, utils.logger._get_log_filename()
+        )
+
+    # clears old dir
+    if os.path.exists(log_file_path):
+        os.remove(log_file_path)
+    if os.path.exists(log_file_path_extended):
+        os.remove(log_file_path_extended)
+
+    my_client = utils.client.create(**CONFIG["my_other_instance"])
+    utils.logger.enable_toolbox_logging(
+        log_to_terminal=False, log_directory=log_file_path
```

Comment:
It also won't be necessary to create the `log_path_extended` variable anymore.  Instead, by passing `tempdir` to `utils.logger.enable_toolbox_logging` as its `log_directory` the file will be created in that directory.
```suggestion
    # clears old dir
    if os.path.exists(log_file_path):
        os.remove(log_file_path)

    my_client = utils.client.create(**CONFIG["my_other_instance"])
    utils.logger.enable_toolbox_logging(
        log_to_terminal=False, log_directory=tempdir
```

Reply:
When I run the new code (w/ no `extended` file path), I get a `IsADirectoryError: [Errno 21] Is a directory` error.
This could just be a local issue, so I will commit your changes below.